examples/asr/speech_to_text_finetune.py

```diff
@@ -0,0 +1,137 @@
+from omegaconf import OmegaConf
+from pytorch_lightning.utilities import rank_zero_only
+
+from nemo.collections.asr.models import ASRModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging, model_utils
+from nemo.utils.exp_manager import exp_manager
+
+
+@hydra_runner(config_path="conf", config_name="speech_to_text_finetune")
+def main(cfg):
+    logging.info(f'Hydra config: {OmegaConf.to_yaml(cfg)}')
+
+    trainer = pl.Trainer(**cfg.trainer)
+    exp_manager(trainer, cfg.get("exp_manager", None))
+
+    if hasattr(cfg, 'init_from_ptl_ckpt') and cfg.init_from_ptl_ckpt is not None:
+        raise NotImplementedError(
+            "Currently for simplicity of single script for all model types, we only support `init_from_nemo_model` and `init_from_pretrained_model`"
+        )
+
+    @rank_zero_only
+    def get_base_model(cfg):
+        asr_model = None
+        nemo_model_path = cfg.get('init_from_nemo_model', None)
+        pretrained_name = cfg.get('init_from_pretrained_model', None)
+        if nemo_model_path is not None and pretrained_name is not None:
+            raise ValueError("Only pass `init_from_nemo_model` or `init_from_pretrained_model` but not both")
+        elif nemo_model_path is None and pretrained_name is None:
+            raise ValueError(
+                "Both `init_from_nemo_model` and `init_from_pretrained_model cannot be None, should pass atleast one of them"
+            )
+        elif nemo_model_path is not None:
+            asr_model = ASRModel.restore_from(restore_path=nemo_model_path)
+        elif pretrained_name is not None:
+            asr_model = ASRModel.from_pretrained(model_name=pretrained_name)
+
+        return asr_model
+
+    asr_model = get_base_model(cfg)
+    vocab_size = asr_model.tokenizer.vocab_size
+
+    # if new tokenizer is provided, use it
+    if hasattr(cfg.model.tokenizer, 'update_tokenizer') and cfg.model.tokenizer.update_tokenizer:
+        decoder = copy.deepcopy(asr_model.decoder)
+        joint_state = copy.deepcopy(asr_model.joint)
+
+        if cfg.model.tokenizer.dir is None:
+            raise ValueError("dir must be specified if update_tokenizer is True")
+        logging.info("Using the tokenizer provided through config")
+        asr_model.change_vocabulary(
+            new_tokenizer_dir=cfg.model.tokenizer.dir, new_tokenizer_type=cfg.model.tokenizer.type
+        )
+        if asr_model.tokenizer.vocab_size != vocab_size:
+            logging.warning(
+                "The vocabulary size of the new tokenizer differs from that of the loaded model. As a result, finetuning will proceed with the new vocabulary, and the decoder will be reinitialized."
+            )
+        else:
+            asr_model.decoder = decoder
+            asr_model.joint = joint_state
+    else:
+        logging.info("Reusing the tokenizer from the loaded model.")
+
+    # Setup Data
+    cfg = model_utils.convert_model_config_to_dict_config(cfg)
```

Comment:
Here onward, these are one liners, probably no need to put into functions. Keep them as is